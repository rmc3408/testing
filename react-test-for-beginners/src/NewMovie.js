import React, { Component } from 'react'
import MovieForm from './MovieForm'

class NewMovie extends Component {
    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(txt) {
        //e.preventDefault();
        console.log(txt);
    }

    render() {
        return (
            <div>
                <h1 data-testid='title'>New Movie</h1>
                <MovieForm subForm={this.handleSubmit}/>
            </div>
        )
    }
}
export default NewMovie;