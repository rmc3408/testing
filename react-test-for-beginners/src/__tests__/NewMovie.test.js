import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react';
import NewMovie from '../NewMovie'

afterEach(cleanup);


test('<NewMovie Snapshot/>', () => {
    const { container } = render(<NewMovie />);
    expect(container.firstChild).toMatchSnapshot();
})

test('<NewMovie True/>', () => {
    const { debug, queryByTestId } = render(<NewMovie />);
    const form = queryByTestId('form'); //Can exist or not. 
    expect(form).toBeTruthy();
})


test('<NewMovie Title/>', () => {
    const { getByTestId } = render(<NewMovie />);
    const h1 = getByTestId('title');
    expect(h1.innerHTML).toBe('New Movie');
    //debug();
})


