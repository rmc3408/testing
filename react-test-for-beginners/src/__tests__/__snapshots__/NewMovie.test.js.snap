// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie Snapshot/> 1`] = `
<div>
  <h1
    data-testid="title"
  >
    New Movie
  </h1>
  <div>
    <form
      data-testid="form"
    >
      <label
        for="text"
      >
        Name
      </label>
      <input
        id="text"
        type="text"
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</div>
`;
